staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/metrics.go

```diff
@@ -106,65 +106,65 @@ var (
 		[]string{priorityLevel, flowSchema},
 	)
 	// PriorityLevelExecutionSeatsObserverGenerator creates observers of seats occupied throughout execution for priority levels
-	PriorityLevelExecutionSeatsObserverGenerator = NewSampleAndWaterMarkHistogramsGenerator(clock.RealClock{}, time.Millisecond,
+	PriorityLevelExecutionSeatsObserverGenerator = NewSampleAndCountHistogramsGenerator(clock.RealClock{}, time.Millisecond*10,
 		&compbasemetrics.HistogramOpts{
 			Namespace:      namespace,
 			Subsystem:      subsystem,
 			Name:           "priority_level_seat_count_samples",
-			Help:           "Periodic observations of number of seats occupied for any stage of execution (but only initial stage for WATCHes)",
+			Help:           "Periodic observations of utilization of seats occupied for any stage of execution (but only initial stage for WATCHes)",
 			Buckets:        []float64{0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1},
 			ConstLabels:    map[string]string{phase: "executing"},
 			StabilityLevel: compbasemetrics.ALPHA,
 		},
-		&compbasemetrics.HistogramOpts{
+		&compbasemetrics.CounterOpts{
 			Namespace:      namespace,
 			Subsystem:      subsystem,
-			Name:           "priority_level_seat_count_watermarks",
-			Help:           "Watermarks of the number of seats occupied for any stage of execution (but only initial stage for WATCHes)",
-			Buckets:        []float64{0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1},
+			Name:           "priority_level_seat_count_band_secs",
+			Help:           "Accumulated seconds spent in bands of utilization of seats occupied for any stage of execution (but only initial stage for WATCHes)",
 			ConstLabels:    map[string]string{phase: "executing"},
 			StabilityLevel: compbasemetrics.ALPHA,
 		},
+		[]float64{0.9, 1},
```

Comment:
I was thinking about it, and how about 0.5, 0.9 and 0.99?

Reply:
Can you explain your thinking?

The virtue of 1 is that it tells us how much time was spent completely saturated.  For a priority level with a concurrency limit of 100 or more, that is very different from --- and, I think, more interesting than --- the amount of time with at least 99% utilized.

Maybe 0.9 is pretty boring, it is really unlikely that a lot of time the utilization will be in [0.9, 1) without this showing up in the samples.

Reply:
I'm very confused by this, you're replacing a histogram with a counterVec w/ explicit buckets? Why not just reduce the buckets in the existing histogram? The only difference between a counterVec and a histogram is the aggregate metrics you get with a histogram (you get two additional summary metrics).

Reply:
The reason is that histogram/summary compute quantiles from reported observations. This is not what we precisely care about. Because what we want is to say what percentage of the "real time" we were X% saturated.


@MikeSpreitzer - value of "1" was kind-of special until each request was occupying 1 seat. Now, it's no longer that special, because we may not be able to consume more requests even with occupancy less than 1.
Now - as a cluster operator I would like to be able to use those metrics not just to signal me we're out of capacity, but also to be able to tune them e.g. on organic growth of the load. I don't have very strong preference about exact numbers, but 0.5 is kind of useful value, and 0.9 and 0.99 are "stop-the-gaps".